### ISSUE TITLE
Setup Wizard - 'Responsibilities of administrator' is shown twice when importing from a device with multiple facilities

### ISSUE THREAD

**Issue by pcenov** [learningequality/kolibri#12894]
## Observed behavior
The 'Responsibilities of administrator' page is shown twice when importing from a device with one or multiple facilities:

https://github.com/user-attachments/assets/c2b39ded-c5c3-4f4f-9c18-5a5861d6fb10

The issue is extant in both 0.17.3 and 0.18.

## Expected behavior

The 'Responsibilities of administrator' page should be displayed only once.

## Steps to reproduce the issue

1. Install the build asset from https://github.com/learningequality/kolibri/pull/12893 on one device.
2. Install it on another device.
3. On the first device setup a device with one or multiple facilities
4. On the second device go through the 'Import all data from an existing learning facility' flow.

## Logs

[logs.zip](https://github.com/user-attachments/files/17949576/logs.zip)

## Usage Details
Kolibri 0.17.3 and 0.18
Ubuntu 22  - Chrome

**Comment by pcenov** [comment]
@radinamatic

**Comment by MisRob** [comment]
Here's a guide on [Running another Kolibri instance alongside the development server](https://kolibri-dev.readthedocs.io/en/develop/howtos/another_kolibri_instance.html)

**Comment by Jeevansm25** [comment]
Can I take this issue?

**Comment by AlexVelezLl** [comment]
Hi @Jeevansm25! For sure! I will assign this issue to you. Please be aware that this issue is in our `setup_wizard` plugin. To reproduce this issue you need to have a fresh start of Kolibri so you will need to create new folders and override your `KOLIBRI_HOME`, or remove your default KOLIBRI_HOME folder each time you finished the setup and want to reproduce it again. To have another Kolibri instance running for the import step you can check MisRob's comment. Let us know if you have any questions.

**Comment by Jeevansm25** [comment]
I have identified the issue and corrected the code. Now, I would like to
test the changes. Previously, before correcting the code, I downloaded the
.pex file from the workflow. Could you guide me on how to test the updated
code?

**Comment by MisRob** [comment]
Hi @Jeevansm25, if you follow the guide I posted on running the two instances, one of them is a development server. Open Kolibri from its URL and it should have your latest local updates.

You can consider the other .pex instance just for having the other facility available - generally it shouldn't be problem for it to run on the older version.  Let us know if this work, and if not, please post more details on what exactly you're doing.

**Comment by Jeevansm25** [comment]
> Hi [@Jeevansm25](https://github.com/Jeevansm25), if you follow the guide I posted on running the two instances, one of them is a development server. Open Kolibri from its URL and it should have your latest local updates.
> 
> You can consider the other .pex instance just for having the other facility available - generally it shouldn't be problem for it to run on the older version. Let us know if this work, and if not, please post more details on what exactly you're doing.

After making the changes in the code, do we need to build the project first and then run the Kolibri instances, or can we directly run the instances to see the updates reflected?

Looking forward to your clarification.

**Comment by MisRob** [comment]
@Jeevansm25 if you follow the guide step by step and get to "Run your development server", then upon running `yarn run devserver` or `yarn run devserver-hot`, you only need to update code and you should see the updates reflected without the need to rebuild the .pex or rerun the development server.

**Comment by MisRob** [comment]
@Jeevansm25 if you haven't yet done so, set up your development server https://kolibri-dev.readthedocs.io/en/develop/getting_started.html#getting-started

**Comment by Jeevansm25** [comment]
> [@Jeevansm25](https://github.com/Jeevansm25) if you haven't yet done so, set up your development server https://kolibri-dev.readthedocs.io/en/develop/getting_started.html#getting-started

Of course, I followed all these steps, but I didn't see the changes I made on the development server reflected in the instance running through the .pex file. Could you please clarify this step for me?

**Comment by MisRob** [comment]
> I didn't see the changes I made on the development server reflected in the instance running through the .pex file.

@Jeevansm25 Yes, you won't see the changes on the .pex instance. You'd need to build .pex locally for that. I've never done that but could ask others if needed. However, for the majority of tasks it's not needed, perhaps only in some nuanced cases that have to do something with synchronization. I am not sure if this issue is such a case. Would you tell a bit more about why you think you need this?

**Comment by Jeevansm25** [comment]
> > I didn't see the changes I made on the development server reflected in the instance running through the .pex file.
> 
> [@Jeevansm25](https://github.com/Jeevansm25) Yes, you won't see the changes on the .pex instance. You'd need to build .pex locally for that. I've never done that but could ask others if needed. However, for the majority of tasks it's not needed, perhaps only in some nuanced cases that have to do something with synchronization. I am not sure if this issue is such a case. Would you tell a bit more about why you think you need this?

Thank you for clarifying about the .pex instance! I was testing a synchronization issue where "Responsibilities of administrator" appears twice when importing from a device with multiple facilities. I wanted to confirm if this issue was fixed by my development changes. Since the .pex instance doesn’t reflect the changes, I now understand that I might need to build a new .pex file. Could you guide me if building it locally is a standard process here, or should I explore other approaches?

**Comment by MisRob** [comment]
> I wanted to confirm if this issue was fixed by my development changes. 

@Jeevansm25 I'm trying to understand why you can't confirm it on the instance with the development server - what's the nature of your fix? Hopefully then I can then recommend some alternatives if needed. Regarding rebuilding pex, I need to check on this, will get back to you later.

**Comment by MisRob** [comment]
@Jeevansm25 regarding .pex builds, if it's really needed then my colleague recommends

```
pip install -r requirements/build.txt
make dist
make pex
```

**Comment by MisRob** [comment]
Hi @Jeevansm25, I wanted to mention that [Learning Equality will be closed from December 23 to January 5](https://github.com/learningequality/kolibri/discussions/12956).

**Comment by karankoder** [comment]
Hi @MisRob, I noticed that this issue is still open, and it seems @Jeevansm25 is not actively working on it. Could you please assign this issue to me? I'd like to take it up.

**Comment by Jeevansm25** [comment]
Actually I am working on it. If I can't will notice you.

**Comment by karankoder** [comment]
Hi @Jeevansm25,

Thanks for letting me know! I'll look for another issue to contribute to.

**Comment by Jeevansm25** [comment]
Could we test the updated code using a Jest file? If so, where should we place the Jest file in the project structure? 
[The test ensures the mounted method is only called once.]

**Comment by AlexVelezLl** [comment]
Hey @Jeevansm25! Yes, we do use jest for testing. You can follow our [testing guidelines](https://kolibri-dev.readthedocs.io/en/develop/frontend_architecture/unit_testing.html#naming-conventions).

**Comment by MisRob** [comment]
Hi @Jeevansm25, are you still planning to work on this?

**Comment by Abhishek-Punhani** [comment]
@MisRob  In case @Jeevansm25 is busy in some other engagement , I can work on this issue

**Comment by MisRob** [comment]
Hi @Abhishek-Punhani, I think we could re-assign since we didn't hear back from @Jeevansm25. However since this is a bit more tricky to reproduce, I would ask you to try it out first. I left some more guidance in the comments above. Thanks!

**Comment by Abhishek-Punhani** [comment]
Hi @MisRob, I encountered an issue while debugging where the `this.password`  in the following API call is an empty string despite being entered in the input field.

```javascript
return FacilityImportResource.grantSuperUserPermissions({
  user_id: this.selected.value,
  password: this.password,
});
```  

Additionally, the request was not executed when I added a condition to check for `this.password` before making the API call. Consequently, there was no refresh at step 4, and after continuing, the application redirected to the learn page, but it shows 
No resources available.

**Comment by MisRob** [comment]
Hi @Abhishek-Punhani, thank you for reporting. That sounds like a potential issue, yes. Would you be able to have a look if there is an existing issue for this problem, and if not, open one with steps to reproduce?

**Comment by Abhishek-Punhani** [comment]
@MisRob I couldn't find any existing issue related to this, so I have opened a new issue detailing the problem. Please have a look.
#13103

**Comment by MisRob** [comment]
Wonderful, thank you

**Comment by Abhishek-Punhani** [comment]
Hi @MisRob , I noticed #13103 is closed, so I looked into this issue again. I observed there is no reloading at step 5, but on proceeding at step 5, it loads correctly but says no resources available. Then, on clicking on import channels into your device, It shows the list of channels. I think this issue can be closed; if showing `no available resources`  is not the expected behavior, it can be covered separately.

https://github.com/user-attachments/assets/8fc5db5e-64d1-4c02-94ee-19e368031021

**Comment by Abhishek-Punhani** [comment]
Hi @MisRob , I  noticed #13103 is closed, So I looked into this issue again. I observed no reloading at step 5; it loads correctly and shows the final page. But it says `no resources available`, and clicking on import channels shows the list of channels. I think this issue can be closed now, and if showing `no resources available`  is not the expected behavior, it can covered in a separate issue.

https://github.com/user-attachments/assets/9a9ea1b5-406f-4ee7-9414-ae636c6f1b66

**Comment by pcenov** [comment]
Hi @Abhishek-Punhani - thanks for further looking into this, much appreciated! The incorrect redirect to the Library page where we see "No resources available" has already been filed as [issue ](https://github.com/learningequality/kolibri/issues/13071) and will be fixed separately, so I'm going to close this one.
